launcher/icons/IconList.cpp

```diff
@@ -340,7 +340,13 @@ void IconList::installIcon(const QString& file, const QString& name)
     if (!IconUtils::isIconSuffix(fileinfo.suffix()))
         return;
 
-    QString target = FS::PathCombine(getDirectory(), name.isEmpty() ? fileinfo.fileName() : name);
+    QString targetName = name.isEmpty() ? fileinfo.fileName() : name;
+    // Ensure the target has the correct extension from the source file
+    QFileInfo targetInfo(targetName);
+    if (!IconUtils::isIconSuffix(targetInfo.suffix())) {
+        targetName = targetInfo.completeBaseName() + "." + fileinfo.suffix();
+    }
+    QString target = FS::PathCombine(getDirectory(), targetName);
```

Comment:
I would go with the following suggestion:
```suggestion
    QString targetName;
    auto resolveSuffix = [fileinfo]() {
        QString suffix = fileinfo.suffix();
        return IconUtils::isIconSuffix(suffix) ? suffix : "png";
    };
    if (name.isEmpty()) {
        targetName = fileinfo.completeBaseName() + "." + resolveSuffix();
    } else {
        // Ensure the target has the correct extension from the source file
        QFileInfo targetInfo(name);
        auto suffix = targetInfo.suffix();
        targetName = targetInfo.completeBaseName() + "." + (IconUtils::isIconSuffix(suffix) ? suffix : resolveSuffix());
    }
```
As you do not need to double-check the file again if the name is empty, the result would be the same; the icon will inherit the suffix.
Added also an additional check in the case the imported file suffix is not icon suffix then it should fallback to png.

Reply:
Or without any ifs, but keeping the fallback suffix thing (this does more operations if the suffix is valid than the previous one, as the previous one it would calculate the fallback suffix only on demand)
```suggestion
    const QString sourceSuffix = fileinfo.suffix();
    const QString fallbackSuffix = IconUtils::isIconSuffix(sourceSuffix) ? sourceSuffix : QStringLiteral("png");

    const QFileInfo targetInfo = name.isEmpty() ? fileinfo : QFileInfo{ name };

    const QString targetSuffix = targetInfo.suffix();
    const QString suffix = IconUtils::isIconSuffix(targetSuffix) ? targetSuffix : fallbackSuffix;

    const QString targetName = targetInfo.completeBaseName() + QLatin1Char('.') + suffix;

    QString target = FS::PathCombine(getDirectory(), targetName);
```

Reply:
Use any or maybe a combination